lib/danger_plugin.rb

```diff
@@ -46,22 +47,50 @@ def lint(config = nil)
       else
         markdown offenses_message(files_to_report)
       end
-
     end
 
     private
 
-    def rubocop(files_to_lint, force_exclusion, config_path: nil)
-      base_command = ["rubocop", "-f", "json"]
-      base_command.concat(["--force-exclusion"]) if force_exclusion
-      base_command.concat(["--config", config_path.shellescape]) unless config_path.nil?
+    def rubocop(files_to_lint, force_exclusion, only_report_new_offenses, config_path: nil)
+      base_command = ['rubocop', '-f', 'json']
+      base_command.concat(['--force-exclusion']) if force_exclusion
+      base_command.concat(['--config', config_path.shellescape]) unless config_path.nil?
 
       rubocop_output = `#{'bundle exec ' if File.exist?('Gemfile')}#{base_command.join(' ')} #{files_to_lint}`
 
       return [] if rubocop_output.empty?
 
-      JSON.parse(rubocop_output)['files']
-        .select { |f| f['offenses'].any? }
+      files = JSON.parse(rubocop_output)['files']
+
+      filter_out_offenses(files) if only_report_new_offenses
+
+      files.select { |f| f['offenses'].any? }
+    end
+
+    def filter_out_offenses(files)
+      files.each do |file|
+        added_lines = added_lines(file['path']).to_set
+        file['offenses'].select! do |offense|
+          added_lines.include?(offense['location']['line'])
+        end
+      end
+    end
+
+    def added_lines(path)
+      git.diff_for_file(path)
+         .patch
+         .split("\n@@")
+         .tap(&:shift)
+         .flat_map do |chunk|
+           first_line, *diff = chunk.split("\n")
```

Comment:
Oh cool, I hadn't seen this syntax in Ruby before. It looks similar to JavaScript's object spread?

```js
const [first, ...others] = chunk.split("\n")
```

Neat!